aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -143,50 +143,82 @@ static void check_1d(const Tensor& t, const char* arg, const char* fn) {
 }
 
 Tensor addr(const Tensor& self, const Tensor& vec1, const Tensor& vec2, Scalar beta, Scalar alpha) {
-  check_1d(vec1, "vec1", "addr");
-  check_1d(vec2, "vec2", "addr");
-  Tensor b_self;
-  std::tie(b_self) = expand_size(self, {vec1.size(0), vec2.size(0)}, "addr");
-  return at::_addr(b_self, vec1, vec2, beta, alpha);
+  Tensor result = at::empty({0}, self.options());
+  at::addr_out(result, self, vec1, vec2, beta, alpha);
+  return result;
 }
 
 Tensor& addr_(Tensor& self, const Tensor& vec1, const Tensor& vec2, Scalar beta, Scalar alpha) {
-  check_1d(vec1, "vec1", "addr");
-  check_1d(vec2, "vec2", "addr");
-  return at::_addr_(self, vec1, vec2, beta, alpha);
+  return at::addr_out(self, self, vec1, vec2, beta, alpha);
 }
 
 Tensor& addr_out(Tensor &result, const Tensor& self, const Tensor& vec1, const Tensor& vec2, Scalar beta, Scalar alpha) {
   check_1d(vec1, "vec1", "addr");
   check_1d(vec2, "vec2", "addr");
-  Tensor b_self;
-  std::tie(b_self) = expand_size(self, {vec1.size(0), vec2.size(0)}, "addr_out");
-  return at::_addr_out(result, b_self, vec1, vec2, beta, alpha);
+
```

Comment:
Can this be written more simply as:

```
torch.outer(result, vec1, vec2).mul_(alpha);

if (beta.to<double>() != 0.0) {
  result.add_(self * beta);
}
```

I have a few questions about whether this is possible:

- If self and the outer product of vec1 and vec2 are not the same size, what happens?
  - in particular, what happens if beta is zero and this is the case?
  - what if beta is zero and the result of the outer product can broadcast to the size of self?
- Does this let `outer` and `add` handle the necessary error checks? 

I think the answer is that self has to broadcastable to `outer(vec1, vec2)` if `beta != 0`, but the output size will always be the size of `outer(vec1, vec2)` so this is OK? 

I'm not sure that's the behavior we'd go with if we were implementing addr today, but since we're deprecating it let's do what's simplest while trying to preserve the current behavior.